File: circular queue/circular_queue.py
import numpy as np

class circular_queue() :
    
    def __init__(self,size) :
        self.size = size+1
        self.queue = np.array([None]*(self.size))
        self.front = 0
        self.rear = 0
        
    def isempty(self) :
        if self.front == self.rear :
            return True
        else :
            return False
        
    def isfull(self) :
        if (self.rear+1)%self.size == self.front :
            return True
        else :
            return False
        
    def enqueue(self,element) :
        if self.isfull() :
            print("Queue is full cannot enqueue the element")
            return
        else :
            self.rear = (self.rear+1)%self.size
            self.queue[self.rear] = element
            return "Given element inserted sucessfully"
        
    def dequeue(self) :
        if self.isempty() :
            return "Queue is empty cannot dequeue the element"
        else :
            self.front = (self.front+1)%self.size
            output = str(self.queue[self.front]) + " dequeued"
            return output
        
    def print_queue(self) :
        print("\nPrinting elements of queue :- ")
        temp = self.front
        while temp != self.rear :
            temp = (temp+1)%self.size
            print(self.queue[temp],end="\t")
        print("\n")

# creating a circular queue of size 5            
qu = circular_queue(5)

# inserting 5 elements into the circular queue
qu.enqueue(2)
qu.enqueue(3)
qu.enqueue(7)
qu.enqueue(0)
qu.enqueue(1)
qu.print_queue()

# Trying to insert 6 th element
qu.enqueue(5)

# dequeuing two elements for space 
print(qu.dequeue())
print(qu.dequeue())

# Inserting new elements in free space 
qu.enqueue(5)
qu.enqueue(6)

# printing values present in queue after insertion
qu.print_queue()

File: circular queue/circular_queue.c
#include <stdio.h>
#include <stdlib.h>

struct Queue
{
int size;
int front;
int rear;
int *Q;
};

void create(struct Queue *q,int size)
{
q->size=size+1;
q->front=q->rear=0;
q->Q=(int *)malloc(q->size*sizeof(int));
}

void enqueue(struct Queue *q,int x)
{

if((q->rear+1)%q->size==q->front)
printf("Queue is full cannot enqueue the element\n");

else
{
q->rear=(q->rear+1)%q->size;
q->Q[q->rear]=x;
}

}

int dequeue(struct Queue *q)
{
int x=-1;

if(q->front==q->rear)
printf("Queue is empty cannot dequeue the element\n");

else
{
q->front=(q->front+1)%q->size;
x=q->Q[q->front];
printf("%d dequeued\n",x);
}

return x;
}

void print_queue(struct Queue q)
{
int i=q.front+1;
printf("\nPrinting elements of queue :- \n");
do
{
printf("%d \t",q.Q[i]);
i=(i+1)%q.size;
}while(i!=(q.rear+1)%q.size);

printf("\n\n");
}

int main()
{

// creating a circular queue of size 5
struct Queue q;
create(&q,5);

// inserting 5 elements into the circular queue
enqueue(&q,2);
enqueue(&q,3);
enqueue(&q,7);
enqueue(&q,0);
enqueue(&q,1);
print_queue(q);

// Trying to insert 6 th element
enqueue(&q,5);

// dequeuing two elements for space 
dequeue(&q);
dequeue(&q);

// Inserting new elements in free space 
enqueue(&q,5);
enqueue(&q,6);

// printing values present in queue after insertion
print_queue(q);
printf("\n");

return 0;
}